src/survey/views.py

```diff
@@ -32,11 +33,31 @@ def post(self, request):
         return CreateSurveyController(request, data).process()
 
 
+class SurveysView(generics.GenericAPIView):
+    serializer_class = SurveySerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request, match_id):
+        """Get all surveys."""
+        surveys = Survey.objects.filter(completed_match_id=match_id)
+        return success_response(
+            self.serializer_class(surveys, many=True).data, status.HTTP_200_OK
+        )
+
+    def post(self, request, match_id):
+        """Create a survey."""
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            data = request.data
+        return CreateSurveyController(request, data, match_id).process()
+
+
 class SurveyView(generics.GenericAPIView):
     serializer_class = SurveySerializer
     permission_classes = api_settings.CONSUMER_PERMISSIONS
 
-    def get(self, request, id):
+    def get(self, request, id, match_id):
         """Get survey by id."""
         survey = Survey.objects.filter(id=id)
```

Comment:
this returns a list...

Reply:
i am very confused by the whole checking system here

if you want to returns omething by id, the best solution is just to give

```
if Survey.objects.filter(id=id).exists():
  return success_response ...(serializer... (Survey.objects.get(id=id))...
return failure_response ...
```

Reply:
This way is actually pretty clean and it's clear what is being done, I think we can do a refactor in the future to do it like this - you can ignore my earlier comment where you should check the length to make it more explicit and adopt this style.

Reply:
the code I have here does the same thing but yeah I can change for readability purposes